mlir/test/Conversion/MIGraphXToTosa/mixr-to-tosa-ops.mlir

```diff
@@ -217,6 +253,13 @@ module {
      return %0 : tensor<16xf32>
   }
 
+  // CHECK-LABEL: func.func @func_convert
+  // CHECK: tosa.cast
+  func.func @func_convert(%arg0: tensor<16xf16>) -> tensor<16xf32> {
+    %0 = "migraphx.convert"(%arg0) : (tensor<16xf16>) -> tensor<16xf32>
```

Comment:
Just to give a heads up that they are not actually equivalent. 
 - migraphx.convert doesn't saturate
 - tosa.cast saturate

I'd say this is fine because maybe 99% after a migraphx.convert it has additional op to clamp. In that case it is okay to make clamp a no-op.

Reply:
Ok, so, firstly, I think the bullet points are backwards.
Second, we're only implementing `convert` for float-to-float conversions. Can we get some confirmation that `float -> half` is supposed to saturate instead of rounding to infinity if needed? Because those are some weird semantics.

(since `half -> float` doesn't have this problem)

Reply:
Chat with MIGraphX confirms that their reference implementation isn't quite right for floats.

Reply:
For float->half, 
 - could you double check how TOSA to linalg lowering works? Their unit tests are quite helpful there. 
   - If there's a saturate vs. infinity behavior than it's going to be a problem. Either way, Tosa.cast has a deterministic lowering, and by swapping that to a Tosa.cast may force it to a behavior it didn't expect.
 - BTW are there really float-> half conversions in middle of the model? I'd more or less expect that to happen as part of the quant/dequant lowering

Reply:
This isn't related to quant - this is in fp16 models where you want to go for fp32 on some of the elementwise ops, I think.

`tosa.cast` just lowers to `arith.truncf` 

I'm going to @umangyadav here for our half of the discussion, because, as far as I can tell, MIGraphX's convert on floats is supposed to be `truncf()`

Reply:
>  this is in fp16 models where you want to go for fp32 on some of the elementwise ops

Fair enough! Very reasonable up cast behavior.

Reply:
> MIGraphX's convert on floats is supposed to be truncf() 

By `truncf()` if you mean this  https://en.cppreference.com/w/c/numeric/math/trunc  ? 

MIGraphX' Ref implementation just calculates : `std::min(std::numeric_limits<T>::max(), std::max(std::numeric_limits<T>::min(), T(x)));` where T is any type and x is the input.  Notice that it casts inputs to target_type before clamping. 
This should work for any Floating points as long as it is not INFs or NaNs because comparisons don't work on them.

Reply:
Ah. Yeah, I'm seeing the cast to the target type.

... why are those mins and maxes there, then?

Reply:
By truncf I mean, effectively, LLVM's floating-point truncation instruction, which is probably equivalent to C's `truncf()`

Reply:
> .... why are those mins and maxes there, then? 

Not sure, looks like just casting would do it.